precompiles/EcAddG2.yul

```diff
@@ -0,0 +1,602 @@
+            function fp2ScalarMul(a00, a01, scalar) -> c00, c01 {
+                c00 := montgomeryMul(a00, scalar)
+                c01 := montgomeryMul(a01, scalar)
+            }
+
+            /// @notice Computes the multiplication between two Fp2 elements.
+            /// @dev Algorithm 7 in: https://eprint.iacr.org/2010/354.pdf.
+            /// @param a00, a01 The coefficients of the Fp2 element A.
+            /// @param a00, a01 The coefficients of the Fp2 element B.
+            /// @return c00, c01 The coefficients of the element C = A * B.
+            function fp2Mul(a00, a01, b00, b01) -> c00, c01 {
+                c00 := montgomerySub(montgomeryMul(a00, b00), montgomeryMul(a01, b01))
+                c01 := montgomeryAdd(montgomeryMul(a00, b01), montgomeryMul(a01, b00))
+            }
+
+            /// @notice Computes the negative of a Fp2 elements.
+            /// @param a00, a01 The coefficients of the Fp2 element A.
+            /// @return c00, c01 The coefficients of the element C = -A.
+            function fp2Neg(a00, a01) -> c00, c01 {
+                c00, c01 := fp2Sub(0, 0, a00, a01)
+            }
+
+            /// @notice Computes the inverse of a Fp2 element.
+            /// @dev Algorithm 8 in: https://eprint.iacr.org/2010/354.pdf.
+            /// @param a00, a01 The coefficients of the Fp2 element A.
+            /// @return c00, c01 The coefficients of the element C = A^(-1).
+            function fp2Inv(a00, a01) -> c00, c01 {
+                let t0 := montgomeryMul(a00, a00)
+                let t1 := montgomeryMul(a01, a01)
+                t0 := montgomeryAdd(t0, t1)
+                t1 := montgomeryModularInverse(t0)
+                c00 := montgomeryMul(a00, t1)
+                c01 := montgomerySub(0, montgomeryMul(a01, t1))
+            }
+
+            // FALLBACK
+
+            // Retrieve the coordinates from the calldata
+            let a_x0 := calldataload(0)
+            let a_x1 := calldataload(32)
+            let a_y0 := calldataload(64)
+            let a_y1 := calldataload(96)
+
+            let b_x0 := calldataload(128)
+            let b_x1 := calldataload(160)
+            let b_y0 := calldataload(192)
+            let b_y1 := calldataload(224)
+
+            // Check if points are infinite
+            let aIsInfinity := g2AffinePointIsInfinity(a_x0, a_x1, a_y0, a_y1)
+            let bIsInfinity := g2AffinePointIsInfinity(b_x0, b_x1, b_y0, b_y1)
+
+            if and(aIsInfinity, bIsInfinity) {
+                // Infinity + Infinity = Infinity
+                mstore(0, 0)
+                mstore(32, 0)
+                mstore(64, 0)
+                mstore(96, 0)
+                return(0, 128)
+            }
+
+            if aIsInfinity {
+                // Infinity + B = B
+
+                // Ensure that the coordinates are between 0 and the field order
+                if iszero(and(g2CoordinateIsOnFieldOrder(b_x0, b_x1), g2CoordinateIsOnFieldOrder(b_y0, b_y1))) {
```

Comment:
If we want to keep consistency in the variables case we should replace all the snake-case variable for camel-case ones.